### PR TITLE
Review of access worker service

### PR DESCRIPTION
**Pull request by TordJoranger**
2 new branches created for the purpose of code review.

The code is not finished, but ready for review part one. After cr, pull requests will be made for all changes. 

***All changes needs to be done in master afterwards (or merged, but not sure if possible/smart due to unrelated histories)***

### CODE REVIEW

File: src/QueueReceiver.Core/Models/AccessInfo.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+        [JsonProperty("groupId")]
+        public string PlantOid { get; set; }
+
+        [JsonProperty("members")]
+        public List<Member> Members { get; set; }
+    }
+
+    public class Member
```
Comment (kfaln) [inline src/QueueReceiver.Core/Models/AccessInfo.cs:21]: Move to Member.cs ?

File: src/QueueReceiver.Core/Services/AccessService.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        {
+            _personService = personService;
+            _projectService = projectService;
+            _plantService = plantService;
+            _logger = logger;
+        }
+
+        public async Task HandleRequest(AccessInfo accessInfo)
+        {
+            string plantId = await _plantService.GetPlantId(accessInfo.PlantOid);
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/AccessService.cs]: var

File: src/QueueReceiver.Core/Services/AccessService.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+            if (plantId == null)
+            {
+                _logger.LogInformation($"Group not relevant, removing message from queue");
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/AccessService.cs]: Redundant $

File: src/QueueReceiver.Core/Services/AccessService.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                return;
+            }
+
+            if (MessageHasNoRelevantData(accessInfo))
+            {
+                return;
+            }
+
+           var runningJobs = accessInfo.Members.AsParallel().Select(async member =>
+            {
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/AccessService.cs:46]: fix indentation

File: src/QueueReceiver.Core/Services/AccessService.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                    await GiveAccess(member, plantId);
+                }
+            });
+
+            await Task.WhenAll(runningJobs);
+        }
+
+        private async Task RemoveAccess(Member member, string plantId)
+        {
+            Person? person = await _personService.FindByOid(member.UserOid);
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/AccessService.cs:62]: var

File: src/QueueReceiver.Core/Services/AccessService.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+            if (person == null)
+            {
+                _logger.LogInformation($"Person doesn't exist in db so there is no reson to remove access," +
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/AccessService.cs]: Redundant $. Move text to one line.

File: src/QueueReceiver.Core/Services/AccessService.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                    " removing message from queue");
+                return;
+            }
+            _logger.LogInformation($"Removing access for person with id: {person.Id}, to plant {plantId}");
+            await _projectService.RemoveAccessToPlant(person.Id, plantId);
+        }
+
+        private async Task GiveAccess(Member member, string plantId)
+        {
+            Person person = await _personService.FindOrCreate(member.UserOid);
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/AccessService.cs:78]: var

File: src/QueueReceiver.Core/Services/EntryPointService.cs [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+            _queueClient.RegisterMessageHandler(ProcessMessagesAsync, messageHandlerOptions);
+        }
+
+        private async Task ProcessMessagesAsync(Message message, CancellationToken token)
+        {
+            var accessInfo = JsonConvert.DeserializeObject<AccessInfo>(Encoding.UTF8.GetString(message.Body));
+            _logger.LogInformation($"Processing message : { accessInfo }");
+
+            /**
+             * Injecting here because class is singleton.
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/EntryPointService.cs]: Injecting what?

File: src/QueueReceiver.Core/Services/EntryPointService.cs [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+             * It not possible inject scoped dependiences from a constructor of a singleton
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/EntryPointService.cs]: Typo

File: src/QueueReceiver.Core/Services/EntryPointService.cs [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+            using var scope = _scopeFactory.CreateScope();
+            var accessService =
+                scope.ServiceProvider
+                    .GetRequiredService<IAccessService>();
+
+            await accessService.HandleRequest(accessInfo);
+
+            //TODO consider moving to its own class, to be able to test, 
+            //Locktoken now throws exception in tests as it's internal set (and sealed), and not possible to mock
+            string lockToken = message.SystemProperties.LockToken;
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/EntryPointService.cs:68]: var

File: src/QueueReceiver.Core/Services/EntryPointService.cs [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+            await _queueClient.CompleteAsync(lockToken);
+            _logger.LogInformation($"Message completed successfully");
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/EntryPointService.cs]: Redundant $

File: src/QueueReceiver.Core/Services/GraphService.cs [hunk 12]
```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.Extensions.Configuration;
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/GraphService.cs]: Unused

File: src/QueueReceiver.Core/Services/GraphService.cs [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+using QueueReceiver.Core.Models;
+using QueueReceiver.Core.Settings;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace QueueReceiver.Core.Services
+{
+    public class GraphService : IGraphService
+    {
+        private readonly GraphSettings settings;
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/GraphService.cs:14]: _settings ?

File: src/QueueReceiver.Core/Services/GraphService.cs [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+        private readonly ILogger<GraphService> _logger;
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/GraphService.cs]: Unused

File: src/QueueReceiver.Core/Services/GraphService.cs [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+
+        public GraphService(GraphSettings graphSettings, ILogger<GraphService> logger)
+        {
+            settings = graphSettings;
+            _logger = logger;
+        }
+
+        public async Task<AdPerson> GetPersonByOid(string userOid)
+        {
+            AuthenticationResult auth = await GetAccessToken();
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/GraphService.cs:25]: var

File: src/QueueReceiver.Core/Services/GraphService.cs [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+        private async Task<AuthenticationResult> GetAccessToken()
+        {
+            var authority = settings.Authority;
+            //["AzureAd:Authority"];
+            var graphUrl = settings.GraphUrl;//_config["GraphUrl"];
+            var clientId = settings.ClientId;// _config["AzureAd:ClientId"];
+            var clientSecret = settings.ClientSecret;// _config["Azure:ClientSecret"];
+            var authContext = new AuthenticationContext(authority);
+
+            ClientCredential clientCred = new ClientCredential(clientId, clientSecret);
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/GraphService.cs]: var

File: src/QueueReceiver.Core/Services/PersonService.cs [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+            {
+                person = await _personRepository.AddPerson(
+                                    new Person(adPerson.Username, adPerson.Email)
+                                    {
+                                        Oid = adPerson.Oid,
+                                        FirstName = adPerson.GivenName,
+                                        LastName = adPerson.Surname
+                                    });
+            }
+            await _personRepository.SaveChangesAsync();
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/PersonService.cs:61]: Calling save even when person is found. Should it be inside the if statement ?

File: src/QueueReceiver.Core/Services/ProjectService.cs [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            _personProjectRepository = personProjectRepository;
+            _projectRepository = projectRepository;
+            _personUserGroupRepository = personUserGroupRepository;
+            _userGroupRepository = userGroupRepository;
+        }
+
+        public async Task GiveProjectAccessToPlant(long personId, string plantId)
+        {
+            List<Project> projects = await _projectRepository.GetParentProjectsByPlant(plantId);
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/ProjectService.cs]: var

File: src/QueueReceiver.Core/Services/ProjectService.cs [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+            bool updated = false;
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/ProjectService.cs]: var

File: src/QueueReceiver.Core/Services/ProjectService.cs [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+            projects.ForEach(async project=>
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/ProjectService.cs]: space

File: src/QueueReceiver.Core/Services/ProjectService.cs [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                var projectId = project.ProjectId;
+                var personProject = await _personProjectRepository.GetAsync(projectId, personId);
+                if(personProject == null)
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/ProjectService.cs]: space

File: src/QueueReceiver.Core/Services/ProjectService.cs [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+                    personProject.IsVoided = false;
+                    _personProjectRepository.Update(personProject);
+                    //TODO PersonProjectHistory
+                    updated = true;
+                }
+            });
+
+            if (updated)
+            {
+                long userGroupId = await _userGroupRepository.FindIdByUserGroupName("READ"); //TODO add env variable
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/ProjectService.cs]: var

File: src/QueueReceiver.Core/Services/ServiceLocator.cs [hunk 23]
```diff
@@ -0,0 +1,42 @@
+﻿namespace QueueReceiver.Core.Services
+{
+    using Microsoft.Extensions.DependencyInjection;
+    using QueueReceiver.Core.Interfaces;
```
Comment (kfaln) [inline src/QueueReceiver.Core/Services/ServiceLocator.cs]: Redundant qualifier

File: src/QueueReceiver.CoreTests/Services/EntryPointServiceTests.cs [hunk 24]
```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using QueueReceiver.Core.Services;
+
+namespace QueueReceiver.Core.Services.Tests
```
Comment (kfaln) [inline src/QueueReceiver.CoreTests/Services/EntryPointServiceTests.cs]: What's up with this file... does not seem to exists, or it exists but with different content?

File: src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs [hunk 25]
```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.EntityFrameworkCore;
+using QueueReceiver.Core.Models;
+
+namespace QueueReceiver.Infrastructure.Data
+{
+    public class ApplicationDbContext : DbContext
+    {
+        public ApplicationDbContext(
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs]: Weird formatting, should be:

        public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : base(options)
        {
        }

File: src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs [hunk 26]
```diff
@@ -0,0 +1,53 @@
...
+            DbContextOptions<ApplicationDbContext> options
+        ) : base(options)
+        {
+        }
+
+        public virtual DbSet<Person> Persons { get; set; } = null!;
+        public virtual DbSet<PersonProject> Personprojects { get; set; } = null!;
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs:17]: PersonProjects

File: src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs [hunk 27]
```diff
@@ -0,0 +1,53 @@
...
+        public virtual DbSet<Project> Projects { get; set; } = null!;
+        public virtual DbSet<Plant> Plants { get; set; } = null!;
+        public virtual DbSet<PersonUserGroup> PersonUserGroups { get; set; } = null!;
+        public virtual DbSet<PersonProjectHistory> PersonProjectHistories { get; set; } = null!;
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs:21]: Unused

File: src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs [hunk 28]
```diff
@@ -0,0 +1,53 @@
...
+        public virtual DbSet<UserGroup> UserGroups { get; set; } = null!;
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            base.OnModelCreating(modelBuilder);
+
+            modelBuilder.Entity<PersonProject>()
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Data/ApplicationDbContext.cs]: Consider moving to separate configuration classes, one for each model (google it). Perhaps a bit overkill with the current complexity, but it's nice and clean :)

File: src/QueueReceiver.Infrastructure/Repositories/PersonProjectRepository.cs [hunk 29]
```diff
@@ -0,0 +1,57 @@
...
+                .Include(pp => pp.Project!)
+                .ThenInclude(project => project.Plant)
+                .Where(pp => plantId.Equals(pp.Project!.PlantId)
+                    && personId == pp.PersonId);
+
+            personProjects.ForEachAsync(pp => pp.IsVoided = true);
+            _personProjects.UpdateRange(personProjects);
+        }
+
+        public async Task<int> SaveChangesAsync()
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonProjectRepository.cs]: exp body

File: src/QueueReceiver.Infrastructure/Repositories/PersonProjectRepository.cs [hunk 30]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            return await _context.SaveChangesAsync();
+        }
+
+        public async Task<PersonProject> GetAsync(long projectId, long personId)
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonProjectRepository.cs]: exp body

File: src/QueueReceiver.Infrastructure/Repositories/PersonProjectRepository.cs [hunk 31]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            return await _personProjects.FindAsync(projectId, personId);
+        }
+
+        public void Update(PersonProject personProject)
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonProjectRepository.cs]: exp body

File: src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs [hunk 32]
```diff
@@ -0,0 +1,59 @@
...
+            _persons = _context.Persons;
+        }
+
+        public async Task<Person> AddPerson(Person person)
+        {
+            await _persons.AddAsync(person);
+            return person;
+        }
+
+        public Task<Person> FindByUserEmail(string userEmail)
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs]: exp body

File: src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs [hunk 33]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            return _persons
+                 .SingleOrDefaultAsync(person => 
+                    userEmail.Equals(person.Email, System.StringComparison.OrdinalIgnoreCase));
+        }
+
+        public Task<Person> FindByUsername(string userName)
+        {
+           var shortname = userName.Substring(0, userName.IndexOf('@'));
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs]: shortName + indent

File: src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs [hunk 34]
```diff
@@ -0,0 +1,59 @@
...
+            return _persons
+                .SingleOrDefaultAsync(person =>
+                    userName.Equals(person.UserName, System.StringComparison.OrdinalIgnoreCase)
+                    || shortname.Equals(person.UserName, System.StringComparison.OrdinalIgnoreCase)
+                    );
+        }
+
+        public Task<Person> FindByUserOid(string userOid)
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs]: exp body

File: src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs [hunk 35]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            return _persons
+                .SingleOrDefaultAsync(person => userOid.Equals(person.Oid));
+        }
+
+        public async Task<int> SaveChangesAsync()
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs:45]: exp body

File: src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs [hunk 36]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            return await _context.SaveChangesAsync();
+        }
+
+        public void Update(Person person)
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonRepository.cs:48]: exp body

File: src/QueueReceiver.Infrastructure/Repositories/PersonUserGroupRepository.cs [hunk 37]
```diff
@@ -0,0 +1,26 @@
+﻿using QueueReceiver.Core.Interfaces;
+using QueueReceiver.Core.Models;
+using QueueReceiver.Infrastructure.Data;
+using System.Threading.Tasks;
+
+namespace QueueReceiver.Infrastructure.Repositories
+{
+    public class PersonUserGroupRepository : IPersonUserGroupRepository
+    {
+        private readonly ApplicationDbContext context;
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonUserGroupRepository.cs]: _context

File: src/QueueReceiver.Infrastructure/Repositories/PersonUserGroupRepository.cs [hunk 38]
```diff
@@ -0,0 +1,26 @@
...
+        private readonly DbContextSettings _settings;
+
+        public PersonUserGroupRepository(ApplicationDbContext context, DbContextSettings settings)
+        {
+            this.context = context;
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PersonUserGroupRepository.cs]: redundant

File: src/QueueReceiver.Infrastructure/Repositories/PlantRepository.cs [hunk 39]
```diff
@@ -0,0 +1,33 @@
...
+    public class PlantRepository : IPlantRepository
+    {
+        private readonly DbSet<Plant> _plants;
+
+        public PlantRepository(ApplicationDbContext context)
+        {
+            _plants = context.Plants;
+        }
+
+        public Task<bool> Exists(string oid)
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/PlantRepository.cs]: exp body, this and below

File: src/QueueReceiver.Infrastructure/Repositories/ProjectRepository.cs [hunk 40]
```diff
@@ -0,0 +1,30 @@
...
+using QueueReceiver.Infrastructure.Data;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace QueueReceiver.Infrastructure.Repositories
+{
+    public class ProjectRepository : IProjectRepository
+    {
+        private readonly ApplicationDbContext context;
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/ProjectRepository.cs]: _context

File: src/QueueReceiver.Infrastructure/Repositories/ProjectRepository.cs [hunk 41]
```diff
@@ -0,0 +1,30 @@
...
+
+        public ProjectRepository(ApplicationDbContext context)
+        {
+            this.context = context;
+        }
+
+        public Task<List<Project>> GetParentProjectsByPlant(string plantId)
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/ProjectRepository.cs:20]: exp body

File: src/QueueReceiver.Infrastructure/Repositories/UserGroupRepository.cs [hunk 42]
```diff
@@ -0,0 +1,25 @@
...
+    public class UserGroupRepository : IUserGroupRepository
+    {
+        private readonly ApplicationDbContext _context;
+
+        public UserGroupRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<long> FindIdByUserGroupName(string name)
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/Repositories/UserGroupRepository.cs:18]: exp body

File: src/QueueReceiver.Infrastructure/ServiceCollectionSetup.cs [hunk 43]
```diff
@@ -0,0 +1,57 @@
...
+using QueueReceiver.Core.Interfaces;
+using QueueReceiver.Core.Services;
+using QueueReceiver.Infrastructure.Data;
+using QueueReceiver.Infrastructure.Repositories;
+
+namespace QueueReceiver.Infrastructure
+{
+    public static class ServiceCollectionSetup
+    {
+        public static readonly LoggerFactory _myLoggerFactory =
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/ServiceCollectionSetup.cs]: Naming and formatting.

        public static readonly LoggerFactory MyLoggerFactory =
            new LoggerFactory(new []
            {
                new Microsoft.Extensions.Logging.Debug.DebugLoggerProvider()
            });

File: src/QueueReceiver.Infrastructure/ServiceCollectionSetup.cs [hunk 44]
```diff
@@ -0,0 +1,57 @@
...
+            new LoggerFactory(new[]
+            { new Microsoft.Extensions.Logging.Debug.DebugLoggerProvider() });
+
+        public static void AddDbContext(this IServiceCollection services, IConfiguration configuration) =>
+         services.AddDbContext<ApplicationDbContext>(options =>
```
Comment (kfaln) [inline src/QueueReceiver.Infrastructure/ServiceCollectionSetup.cs]: indentation

File: src/QueueReceiver.Worker/Program.cs [hunk 45]
```diff
@@ -0,0 +1,59 @@
...
+using QueueReceiver.Core.Interfaces;
+using QueueReceiver.Core.Services;
+using QueueReceiver.Core.Services.CleanArchitecture.Core.Services;
+using QueueReceiver.Core.Settings;
+using QueueReceiver.Infrastructure;
+using QueueReceiver.Infrastructure.Data;
+using System.IO;
+using System.Net;
+
+namespace QueueReceiverService
```
Comment (kfaln) [inline src/QueueReceiver.Worker/Program.cs]: namespace should be QueueReceiver.Worker ?

File: src/QueueReceiver.Worker/Worker.cs [hunk 46]
```diff
@@ -0,0 +1,36 @@
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using QueueReceiver.Core.Interfaces;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace QueueReceiverService
```
Comment (kfaln) [inline src/QueueReceiver.Worker/Worker.cs]: namespace should be QueueReceiver.Worker ?

File: src/QueueReceiver.Worker/Program.cs [hunk 47]
```diff
@@ -0,0 +1,59 @@
...
+{
+    public class Program
+    {
+        public Program(IConfiguration configuration)
```
Comment (kfaln) [inline src/QueueReceiver.Worker/Program.cs:17]: exp body

File: src/QueueReceiver.Worker/Program.cs [hunk 48]
```diff
@@ -0,0 +1,59 @@
...
+
+        public static void Main(string[] args)
+        {
+            WebRequest.DefaultWebProxy = new WebProxy("http://www-proxy.statoil.no:80"); //TODO move this to infrastructure and add as variable
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+            .ConfigureAppConfiguration((hostingContext, config) =>
```
Comment (kfaln) [inline src/QueueReceiver.Worker/Program.cs:32]: "config" parameter not used, also overwritten below - also not used.

File: src/QueueReceiver.Worker/Worker.cs [hunk 49]
```diff
@@ -0,0 +1,36 @@
...
+{
+    public class Worker : BackgroundService
+    {
+        private readonly ILogger<Worker> _logger;
+        private readonly IEntryPointService entryPointService;
```
Comment (kfaln) [inline src/QueueReceiver.Worker/Worker.cs]: _entryPointService

File: src/QueueReceiver.Worker/Worker.cs [hunk 50]
```diff
@@ -0,0 +1,36 @@
...
+
+        public Worker(ILogger<Worker> logger, IEntryPointService entryPointService)
+        {
+            _logger = logger;
+            this.entryPointService = entryPointService;
```
Comment (kfaln) [inline src/QueueReceiver.Worker/Worker.cs]: redundant "this"

File: tests/QueueReceiver.UnitTests/Core/ITestableQueueClient1.cs [hunk 51]
```diff
@@ -0,0 +1,11 @@
+﻿using Microsoft.Azure.ServiceBus;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace QueueReceiver.UnitTests.Core
+{
+    internal interface ITestableQueueClient
```
Comment (kfaln) [inline tests/QueueReceiver.UnitTests/Core/ITestableQueueClient1.cs]: not used...

**Review by kfaln — COMMENTED** [review]
Have not reviewed unit tests since this is probably subject to change.

**Comment by TordJoranger** [comment]
Mainly looking for refactoring, rewrite suggestions. Code is subject to change.

**Pull request closed without merging**